src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php

```diff
@@ -528,4 +528,34 @@ public function findCartByTokenValueAndChannel(string $tokenValue, ChannelInterf
             ->getOneOrNullResult()
         ;
     }
+
+    public function sumPaidOrdersByChannelInPeriodGroupedByYearAndMonth(
+        ChannelInterface $channel,
+        \DateTimeInterface $startDate,
+        \DateTimeInterface $endDate,
+    ): array {
+        $queryBuilder = $this->createPaidOrdersSumInChannel($channel);
+
+        return $queryBuilder
+            ->addSelect('YEAR(o.checkoutCompletedAt) AS year')
+            ->addSelect('MONTH(o.checkoutCompletedAt) AS month')
+            ->andWhere('o.checkoutCompletedAt >= :startDate')
+            ->andWhere('o.checkoutCompletedAt <= :endDate')
+            ->setParameter('startDate', $startDate)
+            ->setParameter('endDate', $endDate)
+            ->groupBy('year')
+            ->addGroupBy('month')
+            ->getQuery()
+            ->getArrayResult();
+    }
+
+    private function createPaidOrdersSumInChannel(ChannelInterface $channel): QueryBuilder
```

Comment:
```suggestion
    protected function createPaidOrdersSumInChannelQueryBuilder(ChannelInterface $channel): QueryBuilder
```

1. It's a naming convention to have `QueryBuilder` in  `createAcmeQueryBuilder`.
2. Method should be protected and available for override.